src/components/posts/index.js

```diff
@@ -1,27 +1,49 @@
 import { useEffect, useState } from "react";
 import Post from "../post";
-import { getPosts } from "../../service/apiClient";
 
 const Posts = () => {
-    const [posts, setPosts] = useState([])
+  const [posts, setPosts] = useState([]);
 
-    useEffect(() => {
-        getPosts().then(setPosts)
-    }, [])
+  useEffect(() => {
+    const token = localStorage.getItem('token'); // Adjust according to where your token is stored
 
-    return (
-        <>
-            {posts.map(post => {
-                    return <Post
-                        key={post.id}
-                        name={`${post.author.firstName} ${post.author.lastName}`}
-                        date={post.createdAt}
-                        content={post.content}
-                        comments={post.comments}
-                    />
-            })}
-        </>
-    )
-}
+    fetch('http://localhost:4000/posts', {
```

Comment:
There is an api javascript file for network requests
https://github.com/boolean-uk/team-dev-client-c11/blob/main/src/service/apiClient.js